async.md

```diff
@@ -218,3 +220,100 @@ Example
 			resp.error("caught: "+reason.message);
 		})
 ~~~
+
+# Messaging
+
+This class allows for the execution of async ClearBlade messaging functions provide an MQTT client that will work asynchronously. You can connect to the internal ClearBlade broker or an external broker. 
+
+## MQTT.Message(payload)
+
+The message represents an MQTT message.
+
+* @param {string} payload
+* @constructor
+* @property {string} payload
+* @property {number} qos
+* @property {boolean} retain
+* @property {boolean} duplicate
+
+Example
+
+~~~javascript
+	var msg = new MQTT.Message()
+ 	var msgData = JSON.parse(msg.payload);
+~~~
+
+## MQTT.Client(options)
+
+The client represents an MQTT client connected to a broker.
+If no options are specified, the client connects to the ClearBlade broker.
+
+* @param {object} [options]
+* @param {string} options.address
+* @param {number} [options.port=1883] 
+* @param {string} [options.username] 
+* @param {string} [options.password]
+* @param {string} [options.client_id={randomUUID}]
+* @param {boolean} [options.use_tls=false]
+* @param {object} [options.tls_config]
+* @param {string} options.tls_config.client_cert
+* @param {string} options.tls_config.client_key
+* @param {string} options.tls_config.ca_cert
+* @constructor
+
+Example
+
+~~~javascript
+var client = new MQTT.Client();
+var options = {
+  address: 'platform.clearblade.com',
+  port: 1883,
+  username: "example@clearblade.com",
+  password: "test",
+  use_tls: false,
+  tls_config: {
+    ca_cert: ca_cert,
+  },
+}
+~~~
+
+## Client.subscribe(topic, onMessage)
+
+This function subscribes to a topic in the broker and registers a callback function which is called when a message on that topic is received.
+
+* @param {string} topic
+* @param {function(topic: string, message: Message)} onMessage
+* @returns {Promise}
+
+~~~ javascript
+var client = new MQTT.Client();
+function myStreamService(req, resp){
+  client.subscribe("incoming/data/topic", data)
```

Comment:
The second parameter here (which you called data) is a callback function. Can you define that function above and just make it log the message it receives?